server/api.js

```diff
@@ -349,6 +349,83 @@ router.get("/buyer-tender", async (req, res) => {
 		: res.status(500).send({ code: "SERVER_ERROR" });
 });
 
+router.post("/tender/:tenderId/status", async (req, res) => {
+	const tenderId = parseInt(req.params.tenderId, 10);
+	const newStatus = req.body.status;
+	const validStatuses = ["Active", "In Review", "Closed", "Awarded"];
+	const user = req.user;
+
+	if (!validStatuses.includes(newStatus)) {
+		return res.status(400).send({ code: "INVALID_STATUS" });
+	}
+
+	let client;
+
+	try {
+		client = await pool.connect();
+
+		await client.query("BEGIN");
+
+		const tenderResult = await client.query(
+			"SELECT buyer_id, status FROM tender WHERE id = $1;",
```

Comment:
```suggestion
			"SELECT buyer_id, status FROM tender WHERE id = $1 && buyer_id = $2;",
```

And remove the `if (user.id !== buyerId) {` validation